build-support/bin/ci.py

```diff
@@ -322,32 +349,62 @@ def run_python_tests_v1() -> None:
       green("Contrib unit tests passed.")
 
 
-def run_python_tests_v2() -> None:
-  known_v2_failures_file = "build-support/unit_test_v2_blacklist.txt"
-  with open(known_v2_failures_file, "r") as f:
-    blacklisted_targets = {line.strip() for line in f.readlines()}
-  with travis_section("PythonTestsV1", "Running Python unit tests with V2 test runner"):
+def run_python_tests_v2(*, remote_execution_enabled: bool) -> None:
+
+  def get_blacklisted_targets(path: str) -> Set[str]:
+    return {line.strip() for line in Path(path).read_text().splitlines()}
+
+  blacklisted_v2_targets = get_blacklisted_targets("build-support/unit_test_v2_blacklist.txt")
+  blacklisted_remote_targets = get_blacklisted_targets("build-support/unit_test_remote_blacklist.txt")
+
+  with travis_section("PythonTestsV2Setup", "Setting up Python unit tests with V2 test runner"):
+    all_targets = subprocess.run([
+      "./pants.pex",
+      "--tag=-integration",
+      "--filter-type=python_tests",
+      "filter",
+      "src/python::",
+      "tests/python::",
+    ], stdout=subprocess.PIPE, encoding="utf-8", check=True).stdout.strip().split("\n")
+    v2_targets = set(all_targets) - blacklisted_v2_targets
+    if remote_execution_enabled:
+      remote_execution_targets = v2_targets - blacklisted_remote_targets
```

Comment:
Is this expectation that all new tests will be remote compatible correct?

Reply:
Yes. For now, that likely requires adding new tests to the remote blacklist, because anything that uses an sdist dependency (like `setproctitle` will fail). My main priority now is fixing that issue, which will result in a high likilhood that new tests will already work with remoting.